src/H5T.c

```diff
@@ -3714,13 +3714,8 @@ H5Tdecode(const void *buf)
     if (buf == NULL)
         HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, H5I_INVALID_HID, "empty buffer");
 
-    /* Create datatype by decoding buffer
-     * There is no way to get the size of the buffer, so we pass in
-     * SIZE_MAX and assume the caller knows what they are doing.
-     * Really fixing this will require an H5Tdecode2() call that
-     * takes a size parameter.
-     */
-    if (NULL == (dt = H5T_decode(SIZE_MAX, (const unsigned char *)buf)))
+    /* Create datatype by decoding buffer */
+    if (NULL == (dt = H5T_decode(buf_size, (const unsigned char *)buf)))
```

Comment:
Just a nitpick and not at all important for this PR, but I'd prefer to remove this cast. It's completely useless here (as with many other casts in the library).